src/sample-executors/BaseExecutor.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {Owned} from "solmate/src/auth/Owned.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+
+import {ResolvedOrder, SignedOrder} from "../base/ReactorStructs.sol";
+import {IReactorCallback} from "../interfaces/IReactorCallback.sol";
+import {IReactor} from "../interfaces/IReactor.sol";
+import {Multicall} from "./Multicall.sol";
+
+struct PermitData {
+    address token;
+    bytes data;
+}
+
+abstract contract BaseExecutor is IReactorCallback, Multicall, Owned {
+    IReactor public immutable reactor;
+
+    constructor(IReactor _reactor, address _owner) Owned(_owner) {
+        reactor = _reactor;
+    }
+
+    function reactorCallback(ResolvedOrder[] calldata, bytes calldata callbackData) external virtual;
+
+    function execute(SignedOrder memory order, bytes memory callbackData) external payable {
+        reactor.executeWithCallback{value: msg.value}(order, callbackData);
+    }
+
+    function executeBatch(SignedOrder[] memory orders, bytes memory callbackData) external payable {
+        reactor.executeBatchWithCallback{value: msg.value}(orders, callbackData);
+    }
+
+    /// @notice execute a signed 2612-style permit
+    /// the transaction will revert if the permit cannot be executed
+    function permit(PermitData memory permitData) public {
+        (address owner, address spender, uint256 value, uint256 deadline, uint8 v, bytes32 r, bytes32 s) =
+            abi.decode(permitData.data, (address, address, uint256, uint256, uint8, bytes32, bytes32));
```

Comment:
im confused, why abi encode/decode all this instead of just have them as struct fields?

Reply:
since we also need the `token` address too, I thought it would make more sense to have 
```
struct {
token
data (all permit stuff here)
}
```
But yeah could just put everything top level in the struct

Reply:
ugh also was notified that DAI's permit doesn't follow 2612

Reply:
yeah it doesn't :/ 

you could use Permit2Lib it handles special casing for DAI

Reply:
I'd probably do that and then just put all the params in the PermitData struct

Reply:
Added in latest commit